src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1868,9 +1869,13 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                 return;
             }
 
-            // At this point we know that we have either a delegate type or an expression type for the target.
+            if (reason == LambdaConversionResult.MismatchedReturnType)
+            {
+                Error(diagnostics, ErrorCode.ERR_CantConvAnonMethReturnType, syntax, id, targetType);
```

Comment:
The name of the error code suggests that conversions have something to do, but the logic and message are that the types have to *match*. Consider renaming the error code and diagnostic.

Reply:
Never mind. I see that we're using same naming convention for parameter types (`ERR_CantConvAnonMethParams`). Fine as-is then